src/ref/builtin.md

```diff
@@ -1393,6 +1393,26 @@ input elements, the entire input is retained. Examples:
 Etymology: Haskell.
 
 
+## tilde-abbr
+
+```elvish
+tilde-abbr $pwd
+```
+
+Replace occurrences of the user's home directory in its argument with
+a tilde (`~`). Examples:
+
+```elvish-transcript
+~> tilde-abbr $pwd
+▶ '~'
+~> cd ~/Documents
+~/Documents> echo $pwd
+/Users/foo/Documents
+~/Documents> tilde-abbr $pwd
+▶ '~/Documents'
+```
```

Comment:
These examples can be made more explicitly (and not relying on the default prompt's behavior) by explicitly call out the user's home directory and write the argument literally (instead of relying on `$pwd`, and adding an example demonstrating how `tilde-abbr` respects path boundaries:

```
~> echo $E:HOME
/Users/foo
~> tilde-abbr /Users/foo
▶ '~'
~> tilde-abbr /Users/foobar
▶ /Users/foobar
~> tilde-abbr /Users/foo/a/b
▶ '~/a/b'
```